core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -1634,7 +1638,27 @@ aot_emit_object_data_section_info(uint8 *buf, uint8 *buf_end, uint32 *p_offset,
         EMIT_STR(data_section->name);
         offset = align_uint(offset, 4);
         EMIT_U32(data_section->size);
-        EMIT_BUF(data_section->data, data_section->size);
+        if (obj_data->stack_sizes_section_name != NULL
+            && !strcmp(obj_data->stack_sizes_section_name,
+                       data_section->name)) {
+            uint32 ss_offset = obj_data->stack_sizes_offset;
+            uint32 ss_size =
+                obj_data->func_count * sizeof(*obj_data->stack_sizes);
+            LOG_VERBOSE("Replacing stack_sizes in %s section, offset %" PRIu32
+                        ", size %" PRIu32,
+                        obj_data->stack_sizes_section_name, ss_offset, ss_size);
+            if (ss_offset > 0) {
+                EMIT_BUF(data_section->data, ss_offset);
+            }
+            EMIT_BUF(obj_data->stack_sizes, ss_size);
```

Comment:
Seems that `ss_offset + ss_size` may be larger than `data_section->size` according to L1654?
Had better add check here?
```C
EMIT_BUF(obj_data->stack_sizes, ss_size <= data_section->size - ss_offset ? ss_size : data_section->size - ss_offset);
```

Reply:
> Seems that `ss_offset + ss_size` may be larger than `data_section->size` according to L1654? Had better add check here?

no, it can't.

> 
> ```c
> EMIT_BUF(obj_data->stack_sizes, ss_size <= data_section->size - ss_offset ? ss_size : data_section->size - ss_offset);
> ```

i added an assertion instead.